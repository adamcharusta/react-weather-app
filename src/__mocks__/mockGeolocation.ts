import { GeolocationType } from '../types/geolocationTypes';

export const mockGeolocation: GeolocationType = {
  id: 2553604,
  name: 'White House',
  latitude: 33.58831,
  longitude: -7.61138,
  elevation: 27,
  feature_code: 'PPLA',
  country_code: 'MA',
  admin1_id: 11281879,
  admin2_id: 6547294,
  timezone: 'Africa/Casablanca',
  population: 3144909,
  country_id: 2542007,
  country: 'Morocco',
  admin1: 'Casablanca-Settat',
  admin2: 'Casablanca',
};
