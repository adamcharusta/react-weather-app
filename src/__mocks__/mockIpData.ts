import { IpResponseType } from '../types/ipTypes';

export const mockIpData: IpResponseType = {
  ip: '45.138.153.39',
  network: '45.138.152.0/23',
  version: 'IPv4',
  city: 'Gmina Piątek',
  region: 'Łódź Voivodeship',
  region_code: '10',
  country: 'PL',
  country_name: 'Poland',
  country_code: 'PL',
  country_code_iso3: 'POL',
  country_capital: 'Warsaw',
  country_tld: '.pl',
  continent_code: 'EU',
  in_eu: true,
  postal: '99-120',
  latitude: 52.0696,
  longitude: 19.4734,
  timezone: 'Europe/Warsaw',
  utc_offset: '+0100',
  country_calling_code: '+48',
  currency: 'PLN',
  currency_name: 'Zloty',
  languages: 'pl',
  country_area: 312685,
  country_population: 37978548,
  asn: 'AS50411',
  org: 'ITV Media Sp. z o.o.',
};
