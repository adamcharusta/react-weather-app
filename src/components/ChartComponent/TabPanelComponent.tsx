import React from 'react';
import { Box, Typography } from '@mui/material';

type TabPanelComponentPropsType = {
  children?: React.ReactNode;
  index: number;
  value: number;
};

const TabPanelComponent = ({ children, value, index, ...other }: TabPanelComponentPropsType) => {
  return (
    <div
      role='tabpanel'
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box sx={{ p: 3 }}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
};

export default TabPanelComponent;
