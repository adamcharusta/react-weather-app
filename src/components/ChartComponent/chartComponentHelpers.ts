// implementation like lodash.js chunk
export function chunk<T>(array: T[], size: number): T[][] {
  const length = array == null ? 0 : array.length;
  if (!length || size < 1) return [];
  let index = 0;
  let resIndex = 0;
  const result = new Array(Math.ceil(length / size));

  while (index < length) {
    result[resIndex++] = array.slice(index, (index += size));
  }
  return result;
}

export function a11yProps(index: number): { id: string; 'aria-controls': string } {
  return {
    id: `simple-tab-${index}`,
    'aria-controls': `simple-tabpanel-${index}`,
  };
}
