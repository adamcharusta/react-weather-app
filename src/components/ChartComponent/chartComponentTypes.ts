export type ChartDataType = {
  name: string;
  value: number;
};

export type TabPanelDataType = {
  label: string;
  data: ChartDataType[];
};
