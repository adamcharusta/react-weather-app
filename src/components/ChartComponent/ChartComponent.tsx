import React, { useId } from 'react';
import { a11yProps, chunk } from './chartComponentHelpers';
import moment from 'moment-timezone';
import { TimeFormatEnum } from '../../types/timeFormatEnum';
import { TabPanelDataType } from './chartComponentTypes';
import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  Box,
  Tab,
  Tabs,
  Typography,
  useTheme,
} from '@mui/material';
import TabPanelComponent from './TabPanelComponent';
import {
  CartesianGrid,
  Line,
  LineChart,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
} from 'recharts';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';

type ChartComponentPropTypes = {
  hours: Date[];
  data: number[];
  unit: string;
  expanded: boolean;
  setExpanded: () => void;
  title: string;
};

const ChartComponent = ({
  hours,
  data,
  unit,
  setExpanded,
  expanded,
  title,
}: ChartComponentPropTypes) => {
  const id = useId();
  const theme = useTheme();
  const [value, setValue] = React.useState(0);
  const dailyHours = chunk(hours, 24);
  const dailyData = chunk(data, 24);
  const ariaTitle = title.replace(' ', '-').toLowerCase();

  const chartData = dailyHours.map((hours, i) => {
    const res: TabPanelDataType = {
      label: '',
      data: [],
    };

    res.label = moment(hours[0]).format(TimeFormatEnum.DATE);

    res.data = hours.map((h, j) => ({
      name: moment(h).format(TimeFormatEnum.TIME),
      value: dailyData[i][j],
    }));

    return res;
  });

  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setValue(newValue);
  };

  return (
    <Accordion expanded={expanded} onChange={setExpanded}>
      <AccordionSummary
        aria-controls={`${ariaTitle}-content`}
        id={id}
        expandIcon={<ExpandMoreIcon />}
      >
        <Typography variant='h6'>{title}</Typography>
      </AccordionSummary>
      <AccordionDetails sx={{ paddingLeft: 0, paddingRight: 0 }}>
        <Box sx={{ width: '100%' }}>
          <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
            <Tabs
              value={value}
              onChange={handleChange}
              aria-label={`${ariaTitle}-tab`}
              variant='scrollable'
              scrollButtons='auto'
              allowScrollButtonsMobile
            >
              {chartData.map((chart, index) => (
                <Tab key={index} label={chart.label} {...a11yProps(index)} />
              ))}
            </Tabs>
          </Box>

          {chartData.map((chart, index) => (
            <TabPanelComponent key={index} value={value} index={index}>
              <ResponsiveContainer height={250} width='100%'>
                <LineChart data={chart.data} margin={{ right: 25, top: 10 }}>
                  <CartesianGrid strokeDasharray='3 3' />
                  <XAxis dataKey='name' stroke={theme.palette.text.primary} />
                  <YAxis stroke={theme.palette.text.primary} />
                  <Tooltip
                    formatter={(value) => `${value}${unit}`}
                    contentStyle={{
                      backgroundColor: theme.palette.background.default,
                      fontWeight: 'bold',
                    }}
                  />
                  <Line
                    type='monotone'
                    dataKey='value'
                    stroke={theme.palette.text.secondary}
                    activeDot={{ r: 8 }}
                  />
                </LineChart>
              </ResponsiveContainer>
            </TabPanelComponent>
          ))}
        </Box>
      </AccordionDetails>
    </Accordion>
  );
};

export default ChartComponent;
