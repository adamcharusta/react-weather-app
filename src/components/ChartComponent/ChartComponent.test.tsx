import React from 'react';
import { render } from '@testing-library/react';

import ChartComponent from './ChartComponent';
import { mockWeather } from '../../__mocks__/mockWeather';

describe('ChartComponent', () => {
  it('snapshot open', () => {
    const component = render(
      <ChartComponent
        hours={mockWeather.hourly.time}
        data={mockWeather.hourly.temperature_2m}
        unit={mockWeather?.hourly_units.temperature_2m}
        expanded={true}
        setExpanded={jest.fn}
        title='42'
      />,
    );

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('snapshot close', () => {
    const component = render(
      <ChartComponent
        hours={mockWeather.hourly.time}
        data={mockWeather.hourly.temperature_2m}
        unit={mockWeather?.hourly_units.temperature_2m}
        expanded={false}
        setExpanded={jest.fn}
        title='42'
      />,
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
