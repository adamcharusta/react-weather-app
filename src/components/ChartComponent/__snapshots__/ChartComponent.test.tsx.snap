// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartComponent snapshot close 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-controls="42-content"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
      id=":r1:"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          42
        </h6>
      </div>
      <div
        class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby=":r1:"
            class="MuiAccordion-region"
            id="42-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root css-fpkvcf-MuiAccordionDetails-root"
            >
              <div
                class="MuiBox-root css-8atqhb"
              >
                <div
                  class="MuiBox-root css-1gsv261"
                >
                  <div
                    class="MuiTabs-root css-tkn350-MuiTabs-root"
                  >
                    <div
                      class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
                      style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                    />
                    <div
                      class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        aria-label="42-tab"
                        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                        role="tablist"
                      >
                        <button
                          aria-controls="simple-tabpanel-0"
                          aria-selected="true"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-0"
                          role="tab"
                          tabindex="0"
                          type="button"
                        >
                          19-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-1"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-1"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          20-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-2"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-2"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          21-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-3"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-3"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          22-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-4"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-4"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          23-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-5"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-5"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          24-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-6"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-6"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          25-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <span
                        class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                        style="left: 0px; width: 0px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-labelledby="simple-tab-0"
                  id="chart-tabpanel-0"
                  role="tabpanel"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    />
                    <div
                      class="recharts-responsive-container"
                      style="width: 100%; height: 250px; min-width: 0;"
                    />
                    <p />
                  </div>
                </div>
                <div
                  aria-labelledby="simple-tab-1"
                  hidden=""
                  id="chart-tabpanel-1"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-2"
                  hidden=""
                  id="chart-tabpanel-2"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-3"
                  hidden=""
                  id="chart-tabpanel-3"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-4"
                  hidden=""
                  id="chart-tabpanel-4"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-5"
                  hidden=""
                  id="chart-tabpanel-5"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-6"
                  hidden=""
                  id="chart-tabpanel-6"
                  role="tabpanel"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChartComponent snapshot open 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-controls="42-content"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
      id=":r0:"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          42
        </h6>
      </div>
      <div
        class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby=":r0:"
            class="MuiAccordion-region"
            id="42-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root css-fpkvcf-MuiAccordionDetails-root"
            >
              <div
                class="MuiBox-root css-8atqhb"
              >
                <div
                  class="MuiBox-root css-1gsv261"
                >
                  <div
                    class="MuiTabs-root css-tkn350-MuiTabs-root"
                  >
                    <div
                      class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-agdkjh-MuiTabs-scrollbarSize"
                      style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                    />
                    <div
                      class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-69z67c-MuiTabs-scroller"
                      style="margin-bottom: 0px;"
                    >
                      <div
                        aria-label="42-tab"
                        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                        role="tablist"
                      >
                        <button
                          aria-controls="simple-tabpanel-0"
                          aria-selected="true"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-0"
                          role="tab"
                          tabindex="0"
                          type="button"
                        >
                          19-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-1"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-1"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          20-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-2"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-2"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          21-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-3"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-3"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          22-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-4"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-4"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          23-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-5"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-5"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          24-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                        <button
                          aria-controls="simple-tabpanel-6"
                          aria-selected="false"
                          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                          id="simple-tab-6"
                          role="tab"
                          tabindex="-1"
                          type="button"
                        >
                          25-03-2023
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <span
                        class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                        style="left: 0px; width: 0px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  aria-labelledby="simple-tab-0"
                  id="chart-tabpanel-0"
                  role="tabpanel"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    />
                    <div
                      class="recharts-responsive-container"
                      style="width: 100%; height: 250px; min-width: 0;"
                    />
                    <p />
                  </div>
                </div>
                <div
                  aria-labelledby="simple-tab-1"
                  hidden=""
                  id="chart-tabpanel-1"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-2"
                  hidden=""
                  id="chart-tabpanel-2"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-3"
                  hidden=""
                  id="chart-tabpanel-3"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-4"
                  hidden=""
                  id="chart-tabpanel-4"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-5"
                  hidden=""
                  id="chart-tabpanel-5"
                  role="tabpanel"
                />
                <div
                  aria-labelledby="simple-tab-6"
                  hidden=""
                  id="chart-tabpanel-6"
                  role="tabpanel"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
